subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/mutator/AbstractFileChangeMutator.groovy

```diff
@@ -44,14 +45,12 @@ abstract class AbstractFileChangeMutator extends BuildExperimentListenerAdapter
         return "_" + String.valueOf(timestamp) + "_" + counter
     }
 
-    @Override
-    void beforeInvocation(BuildExperimentInvocationInfo invocationInfo) {
-        change(invocationInfo.projectDir)
+    void beforeBuild(BuildContext context) {
+        change(projectDir)
```

Comment:
We should add `projectDir` to `BuildContext` in the profiler. It seems to be a useful thing.

https://github.com/gradle/gradle-profiler/issues/231

Reply:
I've added a PR that does this: https://github.com/gradle/gradle-profiler/pull/232

Can we depend on a snapshot release of Gradle Profiler? If so, I think we should do that and then we don't have to pass `projectDir` around everywhere.